packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.tsx

```diff
@@ -13,195 +13,17 @@ import {
-
-  public static defaultProps = {
-    cellShape: 'circle',
-    disabled: false
-  };
-
-  private _id: string;
-
-  constructor(props: ISwatchColorPickerOptionProps) {
-    super(props);
-
-    this._id = props.id || getId('colorCell');
-  }
-
-  public render() {
-    let {
-      item,
-      id,
-      className,
-      role,
-      selectedIndex,
-      disabled,
-      cellShape,
-      onClick,
-      onHover,
-      onFocus
-    } = this.props;
-    return (
-      <CommandButton
-        id={ id + '-item' + item.index }
-        data-index={ item.index }
-        data-is-focusable={ true }
-        disabled={ disabled }
-        className={ css(className,
-          {
-            ['is-selected ' + styles.cellIsSelected]: (selectedIndex !== undefined && selectedIndex === item.index),
-            ['is-disabled ' + styles.disabled]: disabled
-          }
-        ) }
-        onClick={ this._onClick }
-        onMouseEnter={ this._onMouseEnter }
-        onMouseLeave={ this._onMouseLeave }
-        onFocus={ this._onFocus }
-        role={ role }
-        aria-selected={ selectedIndex !== undefined && (selectedIndex === item.index).toString() }
-        ariaLabel={ item.label && item.label }
-        title={ item.label && item.label }
-      >
-        { this._onRenderOption() }
-      </CommandButton>
-    );
-  }
-
-  @autobind
-  private _onClick() {
-    let {
-        onClick,
-      disabled,
-      item
-      } = this.props;
-
-    if (onClick && !disabled) {
-      onClick(item);
-    }
-  }
-
-  @autobind
-  private _onMouseEnter() {
-    let {
-        onHover,
-      disabled,
-      item
-      } = this.props;
-
-    if (onHover && !disabled) {
-      onHover(item.id, item.color);
-    }
-  }
-
-  @autobind
-  private _onMouseLeave() {
-    let {
-        onHover,
-      disabled
-      } = this.props;
-
-    if (onHover && !disabled) {
-      onHover();
-    }
-  }
-
-  @autobind
-  private _onFocus() {
-    let {
-        onFocus,
-      disabled,
-      item
-      } = this.props;
-
-    if (onFocus && !disabled) {
-      onFocus(item.id, item.color);
-    }
-  }
-
-  /**
-   * Render the core of a color cell
-   * @returns {JSX.Element} - Element representing the core of the item
-   */
-  @autobind
-  private _onRenderOption(): JSX.Element {
-    let {
-        cellShape,
-      item
-      } = this.props;
-
-    // Build an SVG for the cell with the given shape and color properties
-    return (
-      <svg className={ css(styles.svg, cellShape, cellShape === 'circle' ? styles.circle : '') } viewBox='0 0 20 20' fill={ getColorFromString(item.color as string).str } >
-        {
-          cellShape === 'circle' ?
-            <circle cx='50%' cy='50%' r='50%' /> :
-            <rect width='100%' height='100%' />
-        }
-      </svg>
-    );
-  }
-}
-
-export interface ISwatchColorPickerState {
-  selectedIndex?: number;
+class ColorPickerGridCell extends GridCell<IColorCellProps, IGridCellProps<IColorCellProps>> {
```

Comment:
This is how I have seen examples done with generic types on react components. Since the _renderOption needs a GridCell with the IColorPickerProps type, this class allows us to create a component from it. If there is another way to do this, I haven't found it but would be glad to update the code.